pkg/armrpc/asyncoperation/worker/worker.go

```diff
@@ -267,7 +267,12 @@ func (w *AsyncRequestProcessWorker) runOperation(ctx context.Context, message *q
 			result.SetFailed(armErr, false)
 		}
 
-		logger.Info("Operation returned", "success", result.Error == nil, "provisioningState", result.ProvisioningState(), "err", result.Error)
+		// We need the if/else here to prevent a panic inside the logger.
+		if result.Error == nil {
+			logger.Info("Operation returned", "success", "true", "provisioningState", result.ProvisioningState())
+		} else {
+			logger.Info("Operation returned", "success", "false", "provisioningState", result.ProvisioningState(), "err", result.Error)
+		}
```

Comment:
I found this while debugging. It's not breaking anything, but is causing ugliness in the logs.

Reply:
The actual problem is that `v1.ErrorDetails` implements `error` through a value, rather than a pointer. I would like to change that, but it could be a very impactful change.

Reply:
Should we create an issue for this?

Reply:
I was planning to just fix it.